### PR TITLE
Make RealDenseDistance type available within cookbooks.

### PR DESCRIPTION
**Pull request by arianepaola**
This adds the  DenseDistance type and makes it available within cookbooks.

### CODE REVIEW

**Comment by arianepaola** [comment]
@vigsterkr @karlnapf @lgoetz

Please review and merge.

File: examples/meta/generator/targets/cpp.json [hunk 1]
```diff
@@ -45,7 +45,8 @@
         "ShortRealMatrix": "SGMatrix<float32_t>",
         "RealMatrix": "SGMatrix<float64_t>",
         "LongRealMatrix": "SGMatrix<floatmax_t>",
-        "ComplexMatrix": "SGMatrix<complex128_t>"
+        "ComplexMatrix": "SGMatrix<complex128_t>",
+        "DenseDistance": "CDenseDistance<float64_t>"
```
Comment (karlnapf) [inline examples/meta/generator/targets/cpp.json]: I guess we would call this `RealDistance` to follow our old pattern of calling everything float64_t real

Comment (karlnapf) [inline examples/meta/generator/targets/cpp.json]: i.e. RealFeatures

Comment (karlnapf) [inline examples/meta/generator/targets/cpp.json]: due to the name clash, I suggest
`RealDenseDistance`

Comment (arianepaola) [inline examples/meta/generator/targets/cpp.json]: ok. RealDenseDistance :smile: it is

Comment (arianepaola) [inline examples/meta/generator/targets/cpp.json]: @karlnapf updated

**Comment by karlnapf** [comment]
Thanks!
Travis error is unrelated, merging

**Pull request merged**